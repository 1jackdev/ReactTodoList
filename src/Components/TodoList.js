import { React, useState } from "react";
import { v4 as uuidv4 } from "uuid";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";
import "./TodoList.css";

const TodoList = () => {
  const INITIAL_STATE = [
      {
        id: uuidv4(),
        text: "Walk the dog"
      },
      {
        id: uuidv4(),
        text: "Get groceries"
      },
      {
        id: uuidv4(),
        text: "Do the dishes"
      },
  ];
  const [todoList, setTodoList] = useState(INITIAL_STATE);

  const addTodo = (text) => {
    setTodoList((todos) => [...todoList, { id: uuidv4(), text }]);
  };

  function handleRemove(id) {
    const newTodoList = todoList.filter((todo) => todo.id !== id);

    setBoxList(newTodoList);
  }

  return (
    <div className="container">
      <NewTodoForm addTodo={addTodo} />
      <ul className="todo-list">
        {todoList.map(({ id, text }) => (
          <Box key={id} id={id} text={text} handleRemove={handleRemove} />
        ))}
      </ul>
    </div>
  );
};

export default TodoList;
