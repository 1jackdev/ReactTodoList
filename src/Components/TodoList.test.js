import React from "react";
import { render, fireEvent, queryByText } from "@testing-library/react";
import TodoList from "./TodoList";

function addTodo(todoList, text = "walk the dog") {
  const textInput = todoList.getByAltText("todo-input");
  fireEvent.change(textInput, { target: { value: text } });

  const button = todoList.getByText("Add Todo");
  fireEvent.submit(button);
}

it("renders without crashing", () => {
  render(<TodoList />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("adds a new todo", () => {
  const todoList = render(<TodoList />);

  expect(todoList.queryByText("Remove")).not.toBeInTheDocument();

  addTodo(todoList);

  const removeButton = todoList.getByText("Remove");
  expect(removeButton).toBeInTheDocument();

  expect(removeButton.previousSibling).toHaveTextContent("walk the dog");
});

it("can remove a todo", () => {
  const todoList = render(<TodoList />);
  addTodo(todoList);

  const removeButton = todoList.getByText("Remove");
  fireEvent.click(removeButton);

  expect(removeButton).not.toBeInTheDocument();
});
