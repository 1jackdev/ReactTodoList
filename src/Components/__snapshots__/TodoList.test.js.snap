// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <form>
      <label
        for="text"
      >
        Todo
      </label>
      <input
        id="text"
        name="text"
        placeholder="Walk the dog"
        type="text"
        value=""
      />
      <button>
        Add Todo
      </button>
    </form>
    <ul
      class="todo-list"
    >
      <li
        class="todo-container"
      >
        Walk the dog
      </li>
      <button
        class="btn"
        type="button"
      >
        X
      </button>
      <li
        class="todo-container"
      >
        Get groceries
      </li>
      <button
        class="btn"
        type="button"
      >
        X
      </button>
      <li
        class="todo-container"
      >
        Do the dishes
      </li>
      <button
        class="btn"
        type="button"
      >
        X
      </button>
    </ul>
  </div>
</DocumentFragment>
`;
